File: 463. Island Perimeter/main.py
class Solution:
    def islandPerimeter(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        r=len(grid)
        c=len(grid[0])
        
        #matrix=[[0 for j in range(c)] for i in range(r)]
        res=0 
        for i in range(r):
            for j in range(c):
                count=0
                if grid[i][j]==1:
                    if (i-1==-1) or (grid[i-1][j]==0):
                        count+=1
                    if (i+1==r) or (grid[i+1][j]==0):
                        count+=1
                    if (j-1==-1) or (grid[i][j-1]==0):
                        count+=1
                    if (j+1==c) or (grid[i][j+1]==0):
                        count+=1    
                
                res+=count
                
        return res

File: 463. Island Perimeter/main.cpp
class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        int res = 0; 
        int R = grid.size(), C = grid[0].size(); 
        for(int r = 0; r<R; r++)
            for(int c = 0; c<C; c++){
                if(grid[r][c] == 1){
                    if(r-1 == -1 || grid[r-1][c] == 0){
                        res++;
                    }
                    if(r+1 == R || grid[r+1][c] == 0){
                        res++;
                    }
                    if(c-1 == -1 || grid[r][c-1] == 0){
                        res++;
                    }
                    if(c+1 == C || grid[r][c+1] == 0){
                        res++; 
                    } 
                }
            }
        return res; 
    }
};
